RxRelay/Relays+Infallible.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  Relays+Infallible.swift
+//  RxSwift
+//
+//  Created by Mikhail Markin on 18/06/2022.
+//  Copyright © 2022 Krunoslav Zaher. All rights reserved.
+//
+
+public extension BehaviorRelay {
+    /// Convert to an `Infallible`
+    ///
+    /// - returns: `Infallible<Element>`
+    func asInfallible() -> Infallible<Element> {
+        Infallible(self.asObservable())
+    }
+}
+
+public extension PublishRelay {
```

Comment:
Definitely needs some test coverage :)

Reply:
(basically just something that compares the results of the relay to the infallible is "good enough" here).